packages/eslint-plugin/tests/rules/switch-exhaustiveness-check.test.ts

```diff
@@ -0,0 +1,292 @@
+import path from 'path';
+import switchExhaustivenessCheck from '../../src/rules/switch-exhaustiveness-check';
+import { RuleTester } from '../RuleTester';
+
+const rootPath = path.join(process.cwd(), 'tests/fixtures/');
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    tsconfigRootDir: rootPath,
+    project: './tsconfig.json',
+  },
+});
+
+ruleTester.run('switch-exhaustiveness-check', switchExhaustivenessCheck, {
+  valid: [
+    // All branches matched
+    `
+type Day = 'Monday' | 'Tuesday' | 'Wednesday' | 'Thursday' | 'Friday' | 'Saturday' | 'Sunday'
+
+const day = 'Monday' as Day
+let result = 0
+
+switch (day) {
+  case 'Monday': {
+     result = 1
+     break
+  }
+  case 'Tuesday': {
+     result = 2
+     break
+  }
+  case 'Wednesday': {
+     result = 3
+     break
+  }
+  case 'Thursday': {
+     result = 4
+     break
+  }
+  case 'Friday': {
+     result = 5
+     break
+  }
+  case 'Saturday': {
+     result = 6
+     break
+  }
+  case 'Sunday': {
+     result = 7
+     break
+  }
+} 
+`,
+    // Other primitive literals work too
+    `
+type Num = 0 | 1 | 2
+type Bool = true | false
+type Mix = 0 | 1 | 'two' | 'three' | true
+
+function test1(value: Num): number {
+  switch (value) {
+    case 0: return 0
+    case 1: return 1
+    case 2: return 2
+  }
+}
+
+function test2(value: Bool): number {
+  switch (value) {
+    case true: return 1
+    case false: return 0
+  }
+}
+
+function test3(value: Mix): number {
+  switch (value) {
+    case 0: return 0
+    case 1: return 1
+    case 'two': return 2
+    case 'three': return 3
+    case true: return 4
+  }
+}
+`,
```

Comment:
I can't do multiline suggestions 😠, but could you please split this into 3 separate test cases?
I doubt this would ever be a problem, but in case there's ever a regression which breaks one of them, it makes it easier to figure out exactly what broke.